xdsl/passes.py

```diff
@@ -103,6 +103,24 @@ def from_pass_spec(cls: type[ModulePassT], spec: PipelinePassSpec) -> ModulePass
         # instantiate the dataclass using kwargs
         return cls(**arg_dict)
 
+    @classmethod
+    def get_pass_argument_names_and_types(cls: type[ModulePassT]) -> str:
```

Comment:
I'm not sure I get the purpose of this string. My understanding of the goal is the following: for a ModulePass instance, we want the text that we could pass in the command line to get back the same result exactly. So the tests for this might be to create a pass programmatically, get its string, parse it using `from_pass_spec`, and check for equality.

Reply:
or maybe actually getting the spec instead of the string, that way it's structured